python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -352,17 +387,37 @@ def _set_run_tags(self, run_id: str, cluster: str, task_arn: str):
         }
         self._instance.add_run_tags(run_id, tags)
 
-    def build_ecs_tags_for_run_task(self, run, container_context: EcsContainerContext):
-        if any(tag["key"] == "dagster/run_id" for tag in container_context.run_ecs_tags):
-            raise Exception("Cannot override system ECS tag: dagster/run_id")
+    def build_ecs_tags_for_run_task(self, run: DagsterRun, container_context: EcsContainerContext):
+        run_id_tag = "dagster/run_id"
+        if any(tag["key"] == run_id_tag for tag in container_context.run_ecs_tags):
+            raise Exception(f"Cannot override system ECS tag: {run_id_tag}")
+
+        tags_to_propagate = self._get_tags_to_propagate_to_ecs_task(run)
 
         return [
-            {"key": "dagster/run_id", "value": run.run_id},
+            {"key": run_id_tag, "value": run.run_id},
             {"key": "dagster/job_name", "value": run.job_name},
             *container_context.run_ecs_tags,
+            *tags_to_propagate,
         ]
 
-    def _get_run_tags(self, run_id):
+    def _get_tags_to_propagate_to_ecs_task(self, run: DagsterRun) -> List[Dict[str, str]]:
+        # These tags often contain * or + characters which are not allowed in ECS tags.
+        # They don't seem super useful from an observability perspective, so are excluded from the ECS tags
+
+        tags_to_propagate = []
+        if allow_list := (self.propagate_tags or {}).get("allow_list", []):
+            # Add contextual Dagster run tags to ECS tags
+            if allow_list:
```

Comment:
you already have `if allow_list` above so no need to repeat it here

Reply:
brain fart